### ISSUE TITLE
Call: Fix accessibility issues

### ISSUE THREAD

**Issue by ziggabyte** [zetkin/app.zetkin.org#3728]
### The feature

The call interface is what activists use when they call people in their organization in a phone bank

### The bug

There is a new caller UI that has reached ready MVP status, but it is not good in its accessiblilty ☠️ 

@henrycatalinismith went through the interface with a screenreader and tabbing to see how it works and documented it in a screen recording that you can see below. 

## Steps to reproduce

0. Make sure you have a `.env.local` file in the root of the project that has the line `FEAT_CALL:1` in it
1. Go to https://app.dev.zetkin.org/my and click on one of the call assignments 
2. Use tabbing around to get through the flow

### Expected Behaviour

All the things Henry try in the video should work smoothly

### Actual Behaviour

The interface pretty much does not work at all with tabbing or a screen reader

## Screen recording

https://github.com/user-attachments/assets/2ecae7fc-bae1-4c6d-80a3-4e05ce9afa33

## Henry's notes from the screen recording with suggestions for solutions 

00:08 - 00:51
Calling for peace should be a h1
How to call with Zetkin should be a h2
Start calling should be a h3
Get to know the person should be a h3
Make the phonecall should be a h3
Report the call should be a h3
Previous calls should be a h3
Assignment stats should be a h2
Instructions should be a h3

00:56 - 01:04
Visually hidden content is reachable via keyboard navigation

01:11 - 01:20
Visually hidden content is reachable for screen readers.

01:41 - 01:50
This filter UI is a little bit tricky to deliver in a fully WCAG compliant way due to the mishmash of different roles in play with that toolbar, but it's possible to make it much better even if perfection is out of reach given that design. The Already In and Context buttons are popup menu triggers so they should probably have `aria-hasmenu="popup"`. Then the ones like Today and Tomorrow which function more like toggles should definitely have `aria-pressed="true"` and `aria-pressed="false"` depending on whether they're enabled or not. The whole set of buttons should probably be inside `role="toolbar"` too. The clear filters and calendar icon buttons lack an ARIA label entirely, needs e.g. `aria-label="Date"` or something.

01:51 - 01:57
When I clear all filters, focus is "lost". That's why the next thing you see after that is me tabbing like crazy to get back to where I was in the UI. This is due to the disappearance of the clear filters button after pressing, and is a common accessibility bug in this kind of UI. The usual way of addressing these bugs is by programmatically focusing a logical next element. In this case that would be "Already In".

02:04 - 02:11
Same focus bug again when I open the survey.

02:12 - 02:40
I fight with this for a while lol

02:58 - 03:17
I'm able to fill out the post-call report form before actually ending the call due to the same "Visually hidden content is reachable for screen readers" issue as earlier.

Try chucking [inert](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Global_attributes/inert) on one of the wrapper elements around the bits of the UI nobody's meant to be able to reach. It might be enough all on its own to sort that entire category of bug out for relatively little effort.

**Comment by alexandrakoch** [comment]
#3735 should resolve most of the issues mentioned in the transcript (unless I missed something), but at least the aria attributes in the filter UI still needs to be implemented by someone.
